test/src/kernels/GenericValueTest.C

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GenericValueTest.h"
+
+registerMooseObject("MooseTestApp", GenericValueTest);
+registerMooseObject("MooseTestApp", ADGenericValueTest);
+
+template <bool is_ad>
+InputParameters
+GenericValueTestTempl<is_ad>::validParams()
+{
+  return GenericKernel<is_ad>::validParams();
+}
+
+template <bool is_ad>
+GenericValueTestTempl<is_ad>::GenericValueTestTempl(const InputParameters & parameters)
+  : GenericKernel<is_ad>(parameters)
+{
+}
+
+template <bool is_ad>
+GenericReal<is_ad>
+GenericValueTestTempl<is_ad>::computeQpResidual()
+{
+  return -this->_u[this->_qp] * this->_test[this->_i][this->_qp];
+}
+
+template <bool is_ad>
+Real
+GenericValueTestTempl<is_ad>::computeQpJacobian()
+{
+  return -this->_phi[this->_j][this->_qp] * this->_test[this->_i][this->_qp];
+}
```

Comment:
Such beautiful `this->`. @dschwen I'm open to introducing a `using` macro into `GenericKernel` if application developers want it. Otherwise we will leave it up to derivers from `GenericKernel` whether they want to use `this->` or `using`.

Reply:
using all the way. One line of "using" allows copy and paste, and makes it very clear when diffing between files.

Reply:
We have actually already discussed that using macros are the way to go. I think there was some confusing on how to handle the is_ad parameter to switch between using Kernel:: vs. ADKernel:: , but this should be straight forward with `#if`

Reply:
It would be nice to have using, but Daniel S. and I discussed how this will not work for this situation.

Reply:
I don't understand. You can't put
```c++
using GenericKernel<is_ad>::_j;
using GenericKernel<is_ad>::_i;
using GenericKernel<is_ad>::_qp;
```
in your `GenericValueTestTempl` class template definition? We used to do this all the time. See @dschwen 's code here: https://github.com/idaholab/moose/pull/12670/files#diff-9a55a1a0cce8dd0bdb4aa35837ac3773R19

Reply:
Yes.  I think that works.  We were going about it totally different